cmd/cert-checker/main.go

```diff
@@ -230,130 +229,114 @@ func (c *certChecker) checkCert(cert core.Certificate) (problems []string) {
-			fmt.Fprintln(os.Stderr, "--config is required")
-			os.Exit(1)
-		}
-		configBytes, err := ioutil.ReadFile(configPath)
-		cmd.FailOnError(err, "Failed to read config file")
-		var config cmd.Config
-		err = json.Unmarshal(configBytes, &config)
-		cmd.FailOnError(err, "Failed to parse config file")
-
-		stats, err := metrics.NewStatter(config.Statsd.Server, config.Statsd.Prefix)
-		cmd.FailOnError(err, "Failed to create StatsD client")
-		syslogger, err := syslog.Dial("", "", syslog.LOG_INFO|syslog.LOG_LOCAL0, "")
-		cmd.FailOnError(err, "Failed to dial syslog")
-		logger, err := blog.New(syslogger, 0, 0)
-		cmd.FailOnError(err, "Failed to construct logger")
-		err = blog.Set(logger)
-		cmd.FailOnError(err, "Failed to set audit logger")
-
-		if connect := c.GlobalString("db-connect"); connect != "" {
-			config.CertChecker.DBConnect = connect
-		}
-		if workers := c.GlobalInt("workers"); workers != 0 {
-			config.CertChecker.Workers = workers
-		}
-		config.CertChecker.UnexpiredOnly = c.GlobalBool("valid-only")
-		config.CertChecker.BadResultsOnly = c.GlobalBool("bad-results-only")
-		if cp := c.GlobalString("check-period"); cp != "" {
-			config.CertChecker.CheckPeriod.Duration, err = time.ParseDuration(cp)
-			cmd.FailOnError(err, "Failed to parse check period")
-		}
+	PA cmd.PAConfig
 
-		// Validate PA config and set defaults if needed
-		cmd.FailOnError(config.PA.CheckChallenges(), "Invalid PA configuration")
-
-		saDbURL, err := config.CertChecker.DBConfig.URL()
-		cmd.FailOnError(err, "Couldn't load DB URL")
-		saDbMap, err := sa.NewDbMap(saDbURL, config.CertChecker.DBConfig.MaxDBConns)
-		cmd.FailOnError(err, "Could not connect to database")
-		go sa.ReportDbConnCount(saDbMap, metrics.NewStatsdScope(stats, "CertChecker"))
-
-		pa, err := policy.New(config.PA.Challenges)
-		cmd.FailOnError(err, "Failed to create PA")
-		err = pa.SetHostnamePolicyFile(config.CertChecker.HostnamePolicyFile)
-		cmd.FailOnError(err, "Failed to load HostnamePolicyFile")
-
-		checker := newChecker(
-			saDbMap,
-			clock.Default(),
-			pa,
-			config.CertChecker.CheckPeriod.Duration,
-		)
-		fmt.Fprintf(os.Stderr, "# Getting certificates issued in the last %s\n", config.CertChecker.CheckPeriod)
+	Statsd cmd.StatsdConfig
+
+	Syslog cmd.SyslogConfig
+}
 
-		// Since we grab certificates in batches we don't want this to block, when it
-		// is finished it will close the certificate channel which allows the range
-		// loops in checker.processCerts to break
+func main() {
+	configFile := flag.String("config", "", "File path to the configuration file for this service")
+	workers := flag.Int("workers", runtime.NumCPU(), "The number of concurrent workers used to process certificates")
+	badResultsOnly := flag.Bool("bad-results-only", false, "Only collect and display bad results")
+	connect := flag.String("db-connect", "", "SQL URI if not provided in the configuration file")
+	cp := flag.String("check-period", "2160h", "How far back to check")
```

Comment:
Can you switch out `flag.String` for `flag.Duration` here so we can avoid the `time.ParseDuration` below.